vendor/README.md

```diff
@@ -0,0 +1,56 @@
+# Vendor Mechanism
```

Comment:
Is is more appropriate to put this into `CONTRIBUTING.md`?

Reply:
It seems what you said is also reasonable. Let me try that as well with amending the commits.